src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -918,26 +913,23 @@ private LLVMValueRef LoadVarAddress(int index, LocalVarKind kind, out TypeDesc t
             }
             else if (kind == LocalVarKind.Local)
             {
-                varBase = GetTotalParameterOffset();
-                GetLocalSizeAndOffsetAtIndex(index, out int localSize, out varOffset);
-                valueType = GetLLVMTypeForTypeDesc(_locals[index].Type);
+                varOffset = GetLocalOffsetAtIndex(index);
                 type = _locals[index].Type;
                 if (varOffset == -1)
                 {
                     Debug.Assert(_localSlots[index].Handle != IntPtr.Zero);
                     return _localSlots[index];
                 }
+                varOffset = varOffset + GetTotalParameterOffset();
             }
             else
             {
-                varBase = GetTotalRealLocalOffset() + GetTotalParameterOffset();
-                GetSpillSizeAndOffsetAtIndex(index, out int localSize, out varOffset);
-                valueType = GetLLVMTypeForTypeDesc(_spilledExpressions[index].Type);
+                varOffset = GetSpillOffsetAtIndex(index, GetTotalRealLocalOffset()) + GetTotalParameterOffset();
```

Comment:
This is the change that actually fixes the problem, passing `GetTotalRealLocalOffset()`